map_core/views.py

```diff
@@ -49,3 +56,61 @@ def getStagingModel(p):
 def scaffoldmakerproxy(p = ''):
     url = 'http://localhost:6565/{0}?{1}'.format(p, str(request.query_string, 'utf-8'))
     return getResponseFromRemote(url)
+
+################
+### flatmaps ###
+################
+
+@map_core_blueprint.route('flatmap/')
+def maps():
+    maps = []
+    for path in pathlib.Path(flatmaps_root).iterdir():
+        if os.path.isdir(path) and os.path.join(flatmaps_root, path, 'index.mbtiles'):
+            mbtiles = os.path.join(flatmaps_root, path, 'index.mbtiles')
+            reader = MBTilesReader(mbtiles)
+            rows = reader._query("SELECT value FROM metadata WHERE name='source';")
+            maps.append({ 'id': path.name, 'source': [row[0] for row in rows][0] })
+    return jsonify(maps)
+
+@map_core_blueprint.route('flatmap/<string:map>/')
+def map(map):
+    filename = os.path.join(flatmaps_root, map, 'index.json')
+    return send_file(filename)
+
+@map_core_blueprint.route('flatmap/<string:map>/style')
+def style(map):
+    filename = os.path.join(flatmaps_root, map, 'style.json')
+    return send_file(filename)
+
+@map_core_blueprint.route('flatmap/<string:map>/annotations')
+def map_annotations(map):
+    mbtiles = os.path.join(flatmaps_root, map, 'index.mbtiles')
+    reader = MBTilesReader(mbtiles)
+    rows = reader._query("SELECT value FROM metadata WHERE name='annotations';")
+    annotations = [row[0] for row in rows]
+    return send_file(io.BytesIO(annotations[0].encode('utf-8')), mimetype='application/json')
```

Comment:
This will return as an internal error when the annotations is empty and it is exactly what I have got when testing with mapv0.3.1.